pkg/api/v1alpha2/edgeconnect/edgeconnect_types.go

```diff
@@ -57,7 +57,8 @@ type EdgeConnectSpec struct { //nolint:revive
 
 	// ServiceAccountName that allows EdgeConnect to access the Kubernetes API
 	// +kubebuilder:default:=dynatrace-edgeconnect
-	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+	// +kubebuilder:validation:Required
+	ServiceAccountName string `json:"serviceAccountName"`
```

Comment:
does having both the `default` and `required` make sense?

if not set -> `default` is used
the `required` check will never be hit

all things considered, IMO the only necessary change is that the `isInvalidServiceName` function should be removed as it's never used (due to the `default`)

Reply:
Yes I pointed that out in the description and hoped, that anyone kind of confirms my thoughts. Thanks, will remove the required tag again.

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3763/commits/07540e5000683e05f15abed6ad07d749e9cb0c18


 Please do not forget to still have a look, maybe you find some validation func I can replace with a tag, that I missed.

Reply:
It would make sense to add a `+kubebuilder:validation:MinLength=1` annotation in my opinion. Otherwise empty values would still be possible as only nil values trigger the default value logic.

To check just set `serviceAccountName: ""`

Reply:
I think a regex pattern matching would be better here, as with `+kubebuilder:validation:MinLength=1` `" "` would still be possible.

With `// +kubebuilder:validation:Pattern=\w` I get

`The EdgeConnect "alkr-ec" is invalid: spec.serviceAccountName: Invalid value: "": spec.serviceAccountName in body should match '\w'
`

`The EdgeConnect "alkr-ec" is invalid: spec.serviceAccountName: Invalid value: " ": spec.serviceAccountName in body should match '\w'
`

But all in all I do not know if we should remove the validation function at all and if it would not be better to just revert the changes again... I am not sure if we can cover everything in an "unhacky" way by using kubebuilder tags. What do you guys think?

Reply:
we can't and shouldn't try to protect from every possible dumb way

if we want to overcomplicate the check then yes, probably better to do it with go code

but the go code you removed would also be fooled by " "

I would go with Gabriel's suggestion of `+kubebuilder:validation:MinLength=1`
And pls check all other `+kubebuilder:default` for `strings` and add `+kubebuilder:validation:MinLength=1` (if it make sense)

Reply:
Okay, thanks for the feedback, will do.

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3763/commits/af7b7b4c03d813ccf51243c23947a13c4035e6c1

did not find other default strings that needed change.


This is the resulting error message:

`The EdgeConnect "alkr-ec" is invalid: spec.serviceAccountName: Invalid value: "": spec.serviceAccountName in body should be at least 1 chars long`

I did not find a way to overwrite this message somehow, so I guess we have to go with it. (Just do not like this message that is why I looked for ways to overwrite)

Reply:
not fully, if someone set it directly to `serviceAccountName: ""` then it would have caught it